concourse/model/traits/release.py

```diff
@@ -328,6 +360,9 @@ def deserialise(raw_asset: dict):
             return dacite.from_dict(
                 data_class=data_class,
                 data=raw_asset,
+                config=dacite.Config(
+                    cast=(enum.Enum,),
```

Comment:
Does this work actually? I think you have to cast to `FileAssetMode` 🤔

Reply:
well, I tested it (and also pass enum.Enum at several other invocations of dacite.from_dict:

```
ccc/grafeas_model.py:129:                cast=[Enum],
concourse/model/traits/component_descriptor.py:285:                cast=(enum.Enum,),
concourse/model/traits/release.py:364:                    cast=(enum.Enum,),
concourse/steps/component_descriptor.mako:332:  config=dacite.Config(cast=(enum.Enum,)),
# ... (there are more, but I think those examples should do) :-)
```

Reply:
hmm, and I wonder how dacite is able to find the correct enum class 🤔

Reply:
well that's easy, is it not? The concrete enum is added as typehint (as for all other types as well). To me, it is strange why dacite will not by default cast enums.

Consider the following code:
```
class E(enum.Enum):
 X = 'x'


@dataclasses.dataclass
class D:
 a: E
```

Then the following expression will create an instance of E:

```
D.__dataclass_fields__['a'].type('x')
```

-> as with all typehints for dataclass-attrs, dacite can simply use them (provided the values _are_ types) to "cast" to enum by invoking them.

So to my understanding, by passing `cast=[Enum]` to dacite, we merely tell it that it _should_ try such a cast. We do not tell it _how_ to cast (that we could to using type_hooks parameter). However, I do not quite understand _why_ dacite will not by default also try enum-casting.